[expo-linking][TS] TS7016: Could not find a declaration file for module 'qs'

### Summary

Our company require a strict  TS check, so we have to use `"skipLibCheck": false` so, when you init a new expo project and run `yarn tsc --noEmit` you will see next error: 

```tsx
node_modules/expo-linking/build/Linking.types.d.ts:1:26 - error TS7016: Could not find a declaration file for module 'qs'. '/home/i/tmp/my-brex-expo-ts-app/node_modules/qs/lib/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/qs` if it exists or add a new declaration (.d.ts) file containing `declare module 'qs';`

1 import { ParsedQs } from 'qs';
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android, iOS, Web

### SDK Version (managed workflow only)

45.0.0

### Environment

  expo-env-info 1.0.4 environment info:
    System:
      OS: Linux 5.4 Linux Mint 20.2 (Uma)
      Shell: 3.5.0 - /usr/bin/fish
    Binaries:
      Node: 14.20.0 - /usr/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 6.14.17 - /usr/bin/npm
      Watchman: 20211205.130157.0 - /usr/local/bin/watchman
    npmPackages:
      expo: ~45.0.0 => 45.0.6 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      eas-cli: 0.54.1
      expo-cli: 5.4.12
    Expo Workflow: managed

### Reproducible demo

```tsx
import { parse } from "expo-linking";

const a = parse("");

a.queryParams.nice.ts.support.expo // no errors :(
```

But if I add a typing manually `yarn add -D @types/qs`, TS will begins show errors:

![image](https://user-images.githubusercontent.com/4661784/179976932-44034908-1fcd-43ad-9aff-1ee369ed2ea8.png)

--- a/packages/expo-linking/package.json
+++ b/packages/expo-linking/package.json
@@ -30,13 +30,13 @@
   "license": "MIT",
   "homepage": "https://docs.expo.dev/versions/latest/sdk/linking",
   "dependencies": {
+    "@types/qs": "^6.5.3",
     "expo-constants": "~13.2.0",
     "invariant": "^2.2.4",
     "qs": "^6.9.1",
     "url-parse": "^1.5.9"
   },
   "devDependencies": {
-    "@types/qs": "^6.5.3",
     "expo-module-scripts": "^2.0.0"
   },
   "jest": {

